### PR TITLE
Working on "organization" APIs implementations and/or refactoring

### PR DESCRIPTION
**Pull request by feiranchen**

### CODE REVIEW

File: octokit/organizations.go [hunk 1]
```diff
@@ -2,13 +2,101 @@ package octokit
...
+	}
+	url, err := uri.Expand(params)
+	if err != nil {
+		return make([]Organization, 0), &Result{Err: err}
+	}
+	result = g.client.get(url, &organizations)
+	return
+}
+
 type Organization struct {
```
Comment (pengwynn) [inline octokit/organizations.go]: I think we'll need a [`Plan` struct][plan] for authenticated requests for users that can see that info.

[plan]: https://developer.github.com/v3/orgs/#response-3

**Comment by pengwynn** [comment]
Good start, @lalalah. Let me know if you need help getting a test groove going. :zap:

**Comment by feiranchen** [comment]
Hey @pengwynn I have added the plan struct. However the following API for editing cannot be implemented because Go does not support Patch and doesn't seem likely that it will in near future.
>https://developer.github.com/v3/orgs/#edit-an-organization

Here's a discussion in the Go Lang group talking about possibilities of implementing Patch.
>https://groups.google.com/forum/#!topic/golang-nuts/GjngdEKsUXA

I can see how this can be done in POST with some changes on the server side tho. Please let me know what you think. I will go ahead and work on something else before I hear back.

**Comment by pengwynn** [comment]
I think the mailing list topic was for the [`ParseForm`][parse-form] method. `NewRequest` definitely accepts `PATCH` since we're [using it in other places][example]. Does that help?

[parse-form]: http://golang.org/src/net/http/request.go?s=23467:23502#L769
[example]: https://github.com/octokit/go-octokit/blob/0000dde833b0f66eb38c6ba071b1845f2ff96084/octokit/issues.go#L46

**Comment by feiranchen** [comment]
Cool. My bad. Turns out Request is defined in every layer of the wrappers and I got caught up tracking them down! :stuck_out_tongue_closed_eyes:

The patch method as well as its test cases; testing both method and body of the patch http call.

NOTE: I think this is a typo:
On the development page https://developer.github.com/v3/orgs/#response-3, I think the *description* field in the response should have be updated to:
>"description": "GitHub, the company."

However the page currently says the following, which I think is wrong:
>"description": "A great organization",

I have designed my test with my understanding. If the developer page is not a typo, please let me know and I'll revert it.

**Comment by pengwynn** [comment]
>  I think the description field in the response should have be updated

The sample responses in the docs are there to show the fields, not so much the values. I'd grab the fixtures you need for your tests via `curl` from the API. :grinning:

**Comment by feiranchen** [comment]
Yep I've been doing that! This is the first time I run into a situation where I don't have the permission to view the fields! Would you help me out?

How does this look otherwise?

**Comment by feiranchen** [comment]
For your convenience, the json fixture that I need for organzation(after update) but couldn't curl is here: https://github.com/octokit/go-octokit/blob/5e68239c4c25c486843cd3c4ae2b161b729ffdee/fixtures/organization_updated.json

**Comment by feiranchen** [comment]
updated per discussion in https://github.com/octokit/go-octokit/pull/98#discussion_r27810077

**Comment by pengwynn** [comment]
@feiranchen This is looking good, I like the recent changes. Anything else to implement here before we merge?

**Comment by feiranchen** [comment]
@pengwynn Thanks for your feed back!
It's all good except the test fixture for this specific task was not curl-ed, but rather from the developer's documentation site. This is because I don't have the permission to curl some of these fields.

For your convenience, the json fixture that I need for organzation but couldn't curl is here: https://github.com/octokit/go-octokit/blob/5e68239c4c25c486843cd3c4ae2b161b729ffdee/fixtures/organization_updated.json

**Comment by pengwynn** [comment]
@feiranchen No problem. I just updated the fixture.

**Comment by pengwynn** [comment]
@feiranchen I think we'll have to update the test after changing the fixture. If you need me to take another look, just add the `needs-review` label back on there.

**Comment by feiranchen** [comment]
Synced with test cases. No high level update.

**Comment by pengwynn** [comment]
:cake:

**Pull request merged**